hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcJoinIndexScanProcessorSupplier.java

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.jdbc;
+
+import com.hazelcast.dataconnection.impl.JdbcDataConnection;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.jet.Traverser;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.impl.processor.TransformBatchedP;
+import com.hazelcast.jet.sql.impl.ExpressionUtil;
+import com.hazelcast.jet.sql.impl.JetJoinInfo;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.security.impl.function.SecuredFunction;
+import com.hazelcast.security.permission.ConnectorPermission;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.JetSqlRow;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.security.Permission;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static com.hazelcast.jet.Traversers.traverseIterable;
+import static com.hazelcast.security.permission.ActionConstants.ACTION_READ;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+public class JdbcJoinIndexScanProcessorSupplier
```

Comment:
the name of the processor is misleading. `IndexScan` for me means that we should use index (BTree index or other) in the DB query. Currently SQL optimizer does not know about indexes on JDBC tables and does not take them into account when calculating cost (eg. hash join vs nested loop join in batch queries). The only index that is implicitly used it primary key.

If this processor handles _only_ primary key access to JDBC table then `JdbcJoinByPrimaryKeyProcessorSupplier` would be better.

Reply:
The processor does not handle only primary keys. It handles the join columns in the JOIN ON clause. How about 
JdbcJoinColumnProcessorSupplier ?

Reply:
@orcunc this hasn't been addressed. Base on the current implementation I thing `JdbcJoinPredicateScanProcessorSupplier` makes most sense. Also, please update all the other `*JoinIndexScan*` to `*JoinPredicate*`.

Reply:
rev : 73f84d92f6b0d3923ef6babf25b8adb6d4f60815 renamed all *JoinIndexScan* as *JoinPredicate*